unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/KernelCommunication/WebSocketCommunication/WebSocketCommunication.cs

```diff
@@ -0,0 +1,235 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.NetworkInformation;
+using DCL;
+using UnityEngine;
+using WebSocketSharp.Server;
+public class WebSocketCommunication : IKernelCommunication
+{
+    WebSocketServer ws;
+    private Coroutine updateCoroutine;
+    private bool requestStop = false;
+    
+    private Dictionary<string, GameObject> bridgeGameObjects = new Dictionary<string, GameObject>();
+    
+    public Dictionary<string, string> messageTypeToBridgeName = new Dictionary<string, string>(); // Public to be able to modify it from `explorer-desktop`
+
+    [System.NonSerialized]
+    public static Queue<DCLWebSocketService.Message> queuedMessages = new Queue<DCLWebSocketService.Message>();
+
+    [System.NonSerialized]
+    public static volatile bool queuedMessagesDirty;
+    
+    public bool isServerReady { get { return ws.IsListening; } }
+
+    private bool CheckAvailableServerPort(int port)
+    {
+        bool isAvailable = true;
+
+        // Evaluate current system tcp connections. This is the same information provided
+        // by the netstat command line application, just in .Net strongly-typed object
+        // form.  We will look through the list, and if our port we would like to use
+        // in our TcpClient is occupied, we will set isAvailable to false.
+        IPGlobalProperties ipGlobalProperties = IPGlobalProperties.GetIPGlobalProperties();
+        IPEndPoint[] tcpConnInfoArray = ipGlobalProperties.GetActiveTcpListeners();
+
+        foreach (IPEndPoint endpoint in tcpConnInfoArray) {
+            if (endpoint.Port == port) {
+                isAvailable = false;
+                break;
+            }
+        }
+
+        return isAvailable;
+    }
+
+    private int SearchUnusedPort()
+    {
+        const int minPort = 5000;
+        const int maxPort = 6000;
+
+        for (int i = minPort; i < maxPort; ++i)
+        {
+            if (CheckAvailableServerPort(i))
+            {
+                return i;
+            }
+        }
+        Debug.LogError("There are not unused ports.");
+        return minPort; // error
+    }
+    public WebSocketCommunication()
```

Comment:
```suggestion

    public WebSocketCommunication()
```
The suggestion is to add a single space.